build2: 0.13.0 -> 0.14.0

r-burns opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change



This is a breaking update (at least for the internal C++ APIs) so the
packages all need to be updated in lockstep.

Misc notes:

* build2
  * Remove unnecessary pkg-config patch
  * Trivially refresh remove-config-store-paths.patch
* libbutl
  * Enable tests now that they work with NDEBUG
* libodb
  * update 2.5.0-b.19 -> 2.5.0-b.21


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

r-burns committed:
build2.bootstrap: 0.13.0 -> 0.14.0

r-burns committed:
build2: 0.13.0 -> 0.14.0

This is a breaking update (at least for the internal C++ APIs) so the
packages all need to be updated in lockstep.

Misc notes:

* build2
  * Remove unnecessary pkg-config patch
  * Trivially refresh remove-config-store-paths.patch
* libbutl
  * Enable tests now that they work with NDEBUG
* libodb
  * update 2.5.0-b.19 -> 2.5.0-b.21

risicle reviewed (approved):
LGTM, `nixpkgs-review` happy for me, nixos x86_64 & macos 10.15

Incidentally, what is it that causes the individual packages to not be build on darwin?

r-burns committed:
build2: enable on all platforms

These are working on darwin, and the upstream developers
even use them on windows.

risicle reviewed (approved):
Confirmed all working now, nixos x86_64 & macos 10.15.

LGTM.
